plugins/aws/sts_provisioner.go

```diff
@@ -97,9 +92,14 @@ func (p STSProvisioner) Description() string {
 }
 
 // ChooseTemporaryCredentialsProvider returns the aws provider that fits the scenario described by the current configuration.
-func ChooseTemporaryCredentialsProvider(awsConfig *confighelpers.Config, providerFactory STSProviderFactory) (aws.CredentialsProvider, error) {
+func ChooseTemporaryCredentialsProvider(awsConfig *confighelpers.Config, providerFactory STSProviderFactory, itemFields map[sdk.FieldName]string) (aws.CredentialsProvider, error) {
```

Comment:
nit: you can consider omitting "Choose" from this function's name, as it does not add any value.

Reply:
Done